src/manifest/__tests__/manifest.test.ts

```diff
@@ -0,0 +1,641 @@
+      uris: ["url1", "url2"],
+    };
+
+    const logSpy = jest.spyOn(log, "info").mockImplementation(jest.fn());
+    const fakePeriod = jest.fn((period) => {
+      return { ...period, id: "foo" + period.id, parsingErrors: [period.id] };
+    });
+    const fakeUpdatePeriodInPlace = jest.fn((oldPeriod, newPeriod) => {
+      Object.keys(oldPeriod).map(key => {
+        delete oldPeriod[key];
+      });
+      oldPeriod.id = newPeriod.id;
+    });
+    jest.mock("../period", () =>  ({ default: fakePeriod }));
+    jest.mock("../update_period", () =>  ({ default: fakeUpdatePeriodInPlace }));
+    const Manifest = require("../manifest").default;
+    const manifest = new Manifest(oldManifestArgs, {});
+    const [oldPeriod1] = manifest.periods;
+
+    const eeSpy = jest.spyOn(manifest, "trigger").mockImplementation(jest.fn());
+
+    const newPeriod1 = { id: "pre0" };
+    const newPeriod2 = { id: "pre1" };
+    const newPeriod3 = { id: "foo1" };
+    const newManifest = {
+        _duration: 13,
+      adaptations: {},
+      availabilityStartTime: 6,
+      baseURL: "test2",
+      getDuration() { return 13; },
+      id: "man2",
+      isLive: true,
+      lifetime: 14,
+      minimumTime: 5,
+      parsingErrors: ["c", "d"],
+      presentationLiveGap: 19,
+      suggestedPresentationDelay: 100,
+      timeShiftBufferDepth: 3,
+      periods: [newPeriod1, newPeriod2, newPeriod3],
+      transport: "foob",
+      uris: ["url3", "url4"],
+    };
+
+    manifest.update(newManifest);
+
+    expect(manifest.periods).toEqual([newPeriod1, newPeriod2, newPeriod3]);
+
+    expect(fakeUpdatePeriodInPlace).toHaveBeenCalledTimes(1);
+    expect(fakeUpdatePeriodInPlace).toHaveBeenCalledWith(oldPeriod1, newPeriod3);
+    expect(eeSpy).toHaveBeenCalledTimes(1);
+    expect(eeSpy).toHaveBeenCalledWith("manifestUpdate", null);
+    // expect(logSpy).toHaveBeenCalledTimes(2);
+    // expect(logSpy).toHaveBeenCalledWith(
+    //   "Manifest: Adding new Period pre0 after update.");
+    // expect(logSpy).toHaveBeenCalledWith(
+    //   "Manifest: Adding new Period pre1 after update.");
+    logSpy.mockRestore();
+    eeSpy.mockRestore();
+  });
+
+  it("should append newer Periods when calling `update`", () => {
+    const oldManifestArgs = {
+      availabilityStartTime: 5,
+      baseURL: "test",
+      duration: 12,
+      id: "man",
+      isLive: false,
+      lifetime: 13,
+      minimumTime: 4,
+      parsingErrors: ["a", "b"],
+      periods: [{ id: "1" }],
+      presentationLiveGap: 18,
+      suggestedPresentationDelay: 99,
+      timeShiftBufferDepth: 2,
+      transportType: "foobar",
+      uris: ["url1", "url2"],
+    };
+
+    const logSpy = jest.spyOn(log, "info").mockImplementation(jest.fn());
+    const fakePeriod = jest.fn((period) => {
+      return { ...period, id: "foo" + period.id, parsingErrors: [period.id] };
+    });
+    const fakeUpdatePeriodInPlace = jest.fn((oldPeriod, newPeriod) => {
+      Object.keys(oldPeriod).map(key => {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Consider using "forEach" instead of "map" as its return value is not being used here. ([typescript:S2201](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS2201))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=439&issues=AWgpECqo_MPaBDn0IVT2&open=AWgpECqo_MPaBDn0IVT2)

[ISSUE_KEY]: AWgpECqo_MPaBDn0IVT2